probe_scraper/runner.py

```diff
@@ -102,19 +120,84 @@ def main(temp_dir, out_dir):
                                                           break_by_channel=False)
 
     # Serialize the probe data to disk.
-    write_probe_data(probes_by_channel, revisions, out_dir)
+    write_moz_central_probe_data(probes_by_channel, revisions, out_dir)
+
+
+def load_http_probes(date, cache_dir="cache", out_dir="output"):
+    repos_probes_data, emails = http_scraper.scrape(date, cache_dir)
+
+    # Parse probe data from files into the form:
+    # <repo_name> -> {
+    #   <date> -> {
+    #     "histogram": {
+    #       <histogram_name>: {
+    #         ...
+    #       },
+    #       ...
+    #     },
+    #     "scalar": {
+    #       ...
+    #     },
+    #   },
+    #   ...
+    # }
+    probes = defaultdict(lambda: defaultdict(lambda: defaultdict(int)))
+    for repo_name, dates in repos_probes_data.iteritems():
```

Comment:
AFAICT accessing this loaded structure has multiple points where it can fail due to invalid structure.
In other places we use a pattern of "load", "check structure", "process data", where "check structure" does a basic schema check.
Can we add a simple schema check here to avoid erroring out unexpectedly or adding lots of detailed error handling?